### PR TITLE
[MNT] Modified test_base_rocketGPU.py

### PR DESCRIPTION
**Pull request by AnonymousCodes911**
Refactored the test funcs to use `numpy.testing.assert_array_almost_equal` for more robust testing.

#### Reference Issues/PRs
fixes : #1248

### CODE REVIEW

**Comment by MatthewMiddlehurst** [comment]
Where are the expected results coming from? If it is the output of the GPU rocket then it is not what the issue requested I believe.

**Review by hadifawaz1999 — CHANGES_REQUESTED** [review]
This is not actually what the issue was mentionning, the idea was to use both Rocket and RocketGPU with the same random seed and on the same dataset and make sure the outputs are the same

**Comment by AnonymousCodes911** [comment]
> Where are the expected results coming from? If it is the output of the GPU rocket then it is not what the issue requested I believe.

I misunderstood the issue, apologies for the same, I assumed we had to set an arbitrary seed and test it accordingly. I 'll update the PR at the earliest.

**Comment by MatthewMiddlehurst** [comment]
This has gotten very confused and should probably start from scratch. If you are confused by the issue contents, please ask for clarification.

There should be:
1. new test function created
2. this function could create and fit a [`Rocket`](https://github.com/aeon-toolkit/aeon/blob/main/aeon/transformations/collection/convolution_based/_rocket.py) and [`ROCKETGPU`](https://github.com/aeon-toolkit/aeon/blob/main/aeon/transformations/collection/convolution_based/rocketGPU/_rocket_gpu.py) transform using the same data, parameters and seed
3. this function should then check if the results are the same

**Comment by AnonymousCodes911** [comment]
> This has gotten very confused and should probably start from scratch. If you are confused by the issue contents, please ask for clarification.
> 
> There should be:
> 
> 1. new test function created
> 2. this function could create and fit a [`Rocket`](https://github.com/aeon-toolkit/aeon/blob/main/aeon/transformations/collection/convolution_based/_rocket.py) and [`ROCKETGPU`](https://github.com/aeon-toolkit/aeon/blob/main/aeon/transformations/collection/convolution_based/rocketGPU/_rocket_gpu.py) transform using the same data, parameters and seed
> 3. this function should then check if the results are the same

Got it thank you for the clarification! Apologies for the hassle and confusion!

File: aeon/transformations/collection/convolution_based/_rocket.py [hunk 1]
```diff
@@ -68,11 +70,19 @@ class Rocket(BaseCollectionTransformer):
         "algorithm_type": "convolution",
     }
 
-    def __init__(self, num_kernels=10_000, normalise=True, n_jobs=1, random_state=None):
+    def __init__(
+        self,
+        num_kernels=10_000,
+        n_filters=1000,
```
Comment (hadifawaz1999) [inline aeon/transformations/collection/convolution_based/_rocket.py:76]: in rocket the num_kernels is itself the n_filters, you do not need to add a n_filters parameter

File: aeon/transformations/collection/convolution_based/rocketGPU/tests/test_base_rocketGPU.py [hunk 2]
```diff
@@ -117,3 +127,36 @@ def test_base_rocketGPU_multivariate():
...
+@pytest.mark.skipif(
+    not _check_soft_dependencies("tensorflow", severity="none"),
+    reason="skip test if required soft dependency not available",
+)
+@pytest.fixture
+def random_seed():
+    return 42
+
+
+@pytest.mark.parametrize("n_channels", [1, 3])
```
Comment (hadifawaz1999) [inline aeon/transformations/collection/convolution_based/rocketGPU/tests/test_base_rocketGPU.py:141]: here you should add, as for the other tests in the same file

`@pytest.mark.skipif(
    not _check_soft_dependencies("tensorflow", severity="none"),
    reason="skip test if required soft dependency not available",
)`

File: aeon/transformations/collection/convolution_based/rocketGPU/tests/test_base_rocketGPU.py [hunk 3]
```diff
@@ -117,3 +127,36 @@ def test_base_rocketGPU_multivariate():
...
     # check all ppv values are >= 0
     assert (X_transform[:, 0] >= 0).sum() == len(X)
+
+
+@pytest.mark.skipif(
+    not _check_soft_dependencies("tensorflow", severity="none"),
+    reason="skip test if required soft dependency not available",
+)
+@pytest.fixture
+def random_seed():
```
Comment (hadifawaz1999) [inline aeon/transformations/collection/convolution_based/rocketGPU/tests/test_base_rocketGPU.py:137]: not sure where this is used, if its not used it should be removed

File: aeon/transformations/collection/convolution_based/rocketGPU/tests/test_base_rocketGPU.py [hunk 4]
```diff
@@ -117,3 +127,36 @@ def test_base_rocketGPU_multivariate():
...
+    if n_channels == 1:
+        X, _ = make_example_2d_numpy()
+    else:
+        X, _ = make_example_3d_numpy()
+    X = X[:n_instances]
+
+    n_filters = 100
+    random_seed = 42
+
+    rocket_cpu = Rocket(n_filters=n_filters, random_state=random_seed)
```
Comment (hadifawaz1999) [inline aeon/transformations/collection/convolution_based/rocketGPU/tests/test_base_rocketGPU.py:154]: could you also set the parameter `normalise` in Rocket to false ? this way we are sure both rocket cpu and gpu take the same version of the dataset

**Review by hadifawaz1999 — CHANGES_REQUESTED** [review]
just some comments that should be helpful

File: aeon/transformations/collection/convolution_based/rocketGPU/tests/test_base_rocketGPU.py (hunk 3 above)
Comment (AnonymousCodes911) [reply aeon/transformations/collection/convolution_based/rocketGPU/tests/test_base_rocketGPU.py:137]: image
It was showing this when i tried without this, so i added the fixture is there a way to bypass this?

File: aeon/transformations/collection/convolution_based/_rocket.py (hunk 1 above)
Comment (AnonymousCodes911) [reply aeon/transformations/collection/convolution_based/_rocket.py]: Yes you are right!, sorry for the oversight

File: aeon/transformations/collection/convolution_based/rocketGPU/tests/test_base_rocketGPU.py (hunk 2 above)
Comment (AnonymousCodes911) [reply aeon/transformations/collection/convolution_based/rocketGPU/tests/test_base_rocketGPU.py:141]: done!, rearranged it

File: aeon/transformations/collection/convolution_based/rocketGPU/tests/test_base_rocketGPU.py (hunk 4 above)
Comment (AnonymousCodes911) [reply aeon/transformations/collection/convolution_based/rocketGPU/tests/test_base_rocketGPU.py]: done

**Comment by AnonymousCodes911** [comment]
after making thee changes, and running `pytest -vv` resulted in : 
image
is this expected? if not how should we tackle this?

**Comment by AnonymousCodes911** [comment]
continued in #1325

**Pull request closed without merging**